WordPress/gutenberg
# Converts paragraphs to headings with keyboard shortcuts

## What?

Closes #11173


## Why?

For feature parity with the classic editor in WordPress but also because they're a great set of shortcuts to have.


## How?


Not sure I am doing it right, because:

- This PR adds a new dependency to the block library package for this implementation. 
- Should the block handle this sort of shortcut or should the editor, or the editor implementation (post/site)?
- Is this repetitive way enough? Should we update how transforms work (in particular make `switchToBlockType` receive some 
attributes ...)?

## Testing Instructions





1. Create a post
2. Write a paragraph
3. Press Ctrl + Alt (Option) + [1-6]
4. Observe magic transform to heading that is the level of [1-6]

## Screenshots



https://user-images.githubusercontent.com/107534/194541040-052f0bd2-f586-4908-afa9-ebeb14407257.mp4


## To do

- [x] Fix some typos in the code
- [x] Make sure only the selected block is transformed
- [x] Add a combo+0 to go back to paragraph
- [x] Register the shortcuts for the heading block as well to switch between levels
The mechanism in the heading block would be different as it's only a "level" change. So maybe a separate PR?
- [x] Refactor the registration/handler into a hook
- [x] Make sure the shortcuts appear in the Keyboard shortcuts help modal
- [x] Figure out what to use for Windows browsers as `access+[1-6]` is for pinned items
- [x] Refactor to remove the changes to RichText
